### ISSUE TITLE
X86_64 - TFLite different inference results depending on whether it was compiled with XNNpack delegate or not

### ISSUE THREAD

**Issue by kruglov-dmitry** [tensorflow/tensorflow#55448]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): **_Yes_**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **_4.19.0-18-amd64 1 SMP Debian 4.19.208-1 (2021-09-29) x86_64 GNU/Linux_**
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: **No**
- TensorFlow installed from (source or binary): **_source_**
- TensorFlow version (use command below): **_r2.5, r2.8_**
- Python version: **_3.7.3_**
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source): **gcc 8.3.0**
- CUDA/cuDNN version: **N/A**
- GPU model and memory: **N/A**

**Describe the current behavior**
Running inference in CPP using tensorflow lite library compiled from source using CMake with default settings with XNNPACK support produce wrong results:
```[0, 0, 1, 0, ]```
 If tflite is sompiled with disabled XNNPACK support (`cmake -D TFLITE_ENABLE_XNNPACK=OFF ..`) - model works as expected.
If inference is running from python with XNNPACK support (based on console output - it prints `INFO: Created TensorFlow Lite XNNPACK delegate for CPU.`) - it is provide expected results.
```[1, 0, 0, 0, ]```
+ if we run inference from python with tensorflow==2.5.0 it doesn't show string above in regards of XNNPACK but produce valid results as well.

**Describe the expected behavior**
TFlite compiled with XNNPACK in cpp produce the same result as library without XNNPACK

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no): no

**Standalone code to reproduce the issue**
https://github.com/kruglov-dmitry/tensorflow_lite_from_cpp_python

Repository contains:
- tflite model and test image to run inference
- cpp source, makefile and build instructions
- python source

**Comment by multiverse-tf** [comment]
Currently, we use XNNPACK for TFLite floating-point ops. Comparing with the TFLite that doesn't use XNNPACK, we expect very similar accuracy but won't expect bitwise identical results.

As for your case, what're the input values of the calculation that produces the final integer vector? Do they differ a lot between TFLite w/ XNNPACK and TFLite w/o XNNPACK?

**Comment by kruglov-dmitry** [comment]
Sorry for delay with answer.

> Do they differ a lot between TFLite w/ XNNPACK and TFLite w/o XNNPACK?

w/o XNNPACK:
```
7.89871e+36,
-3.18431e+38,
3.14411e+37,
2.27216e+38,
```

w/ XNNPACK
```
-2.71105e+36,
-inf,
3.25951e+37,
2.47283e+38,
```

same exact model in python:
```
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
[[ 1799.3785    582.42206 -1973.1395  -1173.8969 ]]
```

Model were produced by loading weights of keras model, pop last layer and converting using the same snippet as in repo above.

Model without last layer - [google drive](https://drive.google.com/file/d/1Wk2MlEM_fPTre1AXaqPPplUY5bPYjM1L/view?usp=sharing) ~ 15 mb

**Comment by pkgoogle** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFLite team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the TFLite space.
